zkchannels-crypto/src/pedersen.rs

```diff
@@ -323,74 +283,82 @@ mod test {
 
         // Check validation when the first of the gs in the Pedersen parameters is the identity element
-        let mut bad_params = PedersenParameters::<G1Projective, N>::new(&mut rng);
+        let mut bad_params = PedersenParameters::<G, N>::new(&mut rng);
         let mut gs = bad_params.gs.to_vec();
-        gs[0] = G1Projective::identity();
+        gs[0] = G::identity();
         bad_params.gs = Box::try_from(gs.into_boxed_slice()).unwrap();
         let ser_params = bincode::serialize(&bad_params).unwrap();
-        assert!(bincode::deserialize::<PedersenParameters<G1Projective, N>>(&ser_params).is_err());
+        assert!(bincode::deserialize::<PedersenParameters<G, N>>(&ser_params).is_err());
 
         // Check validation when the last of the gs in the Pedersen parameters is the identity element
-        let mut bad_params = PedersenParameters::<G1Projective, N>::new(&mut rng);
+        let mut bad_params = PedersenParameters::<G, N>::new(&mut rng);
         let mut gs = bad_params.gs.to_vec();
         let last_position = gs.len() - 1;
-        gs[last_position] = G1Projective::identity();
+        gs[last_position] = G::identity();
         bad_params.gs = Box::try_from(gs.into_boxed_slice()).unwrap();
         let ser_params = bincode::serialize(&bad_params).unwrap();
-        assert!(bincode::deserialize::<PedersenParameters<G1Projective, N>>(&ser_params).is_err());
+        assert!(bincode::deserialize::<PedersenParameters<G, N>>(&ser_params).is_err());
 
         // Check validation when the random element of the gs in the Pedersen parameters is the identity element
-        let mut bad_params = PedersenParameters::<G1Projective, N>::new(&mut rng);
+        let mut bad_params = PedersenParameters::<G, N>::new(&mut rng);
         let mut gs = bad_params.gs.to_vec();
         let random_position = rng.gen_range(0..gs.len());
-        gs[random_position] = G1Projective::identity();
+        gs[random_position] = G::identity();
         bad_params.gs = Box::try_from(gs.into_boxed_slice()).unwrap();
         let ser_params = bincode::serialize(&bad_params).unwrap();
-        assert!(bincode::deserialize::<PedersenParameters<G1Projective, N>>(&ser_params).is_err());
+        assert!(bincode::deserialize::<PedersenParameters<G, N>>(&ser_params).is_err());
+    }
+
+    macro_rules! test_commitment_mod {
+        ($group:path, [$length:literal]) => {
+            commit_open_success::<$group, $length>();
+            commit_does_not_open_on_wrong_msg::<$group, $length>();
+            commit_does_not_open_on_wrong_bf::<$group, $length>();
+            commit_does_not_open_on_wrong_commit::<$group, $length>();
+            commit_does_not_open_on_random_commit::<$group, $length>();
+            #[cfg(feature = "bincode")]
+            serialize_deserialize_pedersen_params::<$group, $length>();
+        };
+
+        ($group:path, [$length:literal, $($rest:literal),*]) => {
```

Comment:
This is very cool, I didn't know you could do functional-style matching in macros like this. Looks great.